app/code/Magento/CmsGraphQl/Model/Resolver/DataProvider/Page.php

```diff
@@ -44,10 +62,32 @@ public function __construct(
      * @return array
      * @throws NoSuchEntityException
      */
-    public function getData(int $pageId): array
+    public function getDataByPageId(int $pageId): array
     {
         $page = $this->pageRepository->getById($pageId);
 
+        return $this->convertPageData($page);
+    }
+
+    /**
+     * @param string $pageIdentifier
+     * @return array
+     */
+    public function getDataByPageIdentifier(string $pageIdentifier): array
```

Comment:
You have added new method `getDataByPageIdentifier`, this should be done in a different way - please check this article: https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/
**section** "Introduction of a method to a class or interface"

I think you can solve "public method removal" and "adding a new public method" issues using this approach:
- create two separate handlers for getting data by id `PageById` and by identifier `PageByIdentifier`.
- mark '\Magento\CmsGraphQl\Model\Resolver\DataProvider\Page:getData' method as a deprecated and inject their usage of `PageById` instead of code duplication.
- In `\Magento\CmsGraphQl\Model\Resolver\Page` should be injected `PageById` and `PageByIdentifier` for getting CmsPage data by page id and identifier.
In this case, you will follow the backward compatibility guides.

Reply:
I've created a new data provider model - `app/code/Magento/CmsGraphQl/Model/Resolver/DataProvider/PageDataProvider.php` instead of make changes in the current one - `app/code/Magento/CmsGraphQl/Model/Resolver/DataProvider/Page.php`.

Please review a new solution.

Thank you!